static-check/src/types/functions/function_input_variable.rs

```diff
@@ -0,0 +1,103 @@
+// Copyright (C) 2019-2020 Aleo Systems Inc.
+// This file is part of the Leo library.
+
+// The Leo library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The Leo library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the Leo library. If not, see <https://www.gnu.org/licenses/>.
+
+use crate::{Attribute, ParameterType, SymbolTable, Type, TypeError};
+use leo_typed::{FunctionInputVariable, Identifier, Span};
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct FunctionInputVariableType {
+    /// Name of function input.
+    pub identifier: Identifier,
+
+    /// Type of function input.
+    pub type_: Type,
+
+    /// The attributes of the function input.
+    pub attributes: Vec<Attribute>,
+
+    /// The span of the function input.
+    pub span: Span,
+}
+
+impl FunctionInputVariableType {
+    ///
+    /// Return a new `FunctionInputVariableType` from a given `FunctionInputVariable`.
+    ///
+    /// Performs a lookup in the given symbol table if the type is user-defined.
+    ///
+    pub fn new(table: &SymbolTable, unresolved: FunctionInputVariable) -> Result<Self, TypeError> {
+        let type_ = Type::new(table, unresolved.type_, unresolved.span.clone())?;
+        let attributes = if unresolved.mutable {
+            vec![Attribute::Mutable]
```

Comment:
a reminder about the option to use something lighter than a `Vec` to carry information on an object's `Attribute`s